### PR TITLE
refactor: Drawer styles properties

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

https://preview-46858-ant-design.surge.sh/components/drawer 可以帮忙点点看有没有兼容问题。

已废弃：

```jsx
        <Drawer
          headerStyle={...}
          bodyStyle={...}
          footerStyle={...}
          drawerStyle={...}
          contentWrapperStyle={...}
          maskStyle={...}
        >
          <p>Some contents...</p>
        </Drawer>
```

新用法：

```jsx
        <Drawer
          styles={{
            header: ...,
            body: ...,
            footer: ...,
            content: ...,
            wrapper: ...,
            mask: ...,
          }}
        >
          <p>Some contents...</p>
        </Drawer>
```

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Drawer added  `styles.wrapper` and deprecated `contentWrapperStyle` `drawerStyle`  `maskStyle` props, simplify dom structure by removing `.ant-drawer-wrapper-body`.        |
| 🇨🇳 Chinese |  Drawer 新增 `styles.wrapper` 并废弃 `contentWrapperStyle` `drawerStyle`  `maskStyle` 属性，并简化移除 `.ant-drawer-wrapper-body`  这一层 dom 结构。       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Review by crazyair — APPROVED** [review]

File: components/drawer/DrawerPanel.tsx [hunk 1]
```diff
@@ -5,20 +5,16 @@ import type { DrawerProps as RCDrawerProps } from 'rc-drawer';
 import useClosable from '../_util/hooks/useClosable';
 import { ConfigContext } from '../config-provider';
 
-export interface DrawerClassNames {
+export interface DrawerClassNames extends NonNullable<RCDrawerProps['classNames']> {
```
Comment (crazyair) [inline components/drawer/DrawerPanel.tsx:8]: classNames 有 null 了？

Comment (afc163) [reply components/drawer/DrawerPanel.tsx:8]: 因为是 classNames?: xxx; 所以有 undefined。

Comment (crazyair) [reply components/drawer/DrawerPanel.tsx:8]: 恩，去掉 ? 有 2 种实现方式
```ts
export interface DrawerClassNames {
    mask?: string;
    wrapper?: string;
    content?: string;
}

export interface DrawerProps {
    classNames?: DrawerClassNames;
}
//
//
type Demo2 = NonNullable<DrawerProps['classNames']>;
//
//
type Demo3 = Required<DrawerProps>['classNames'];
```

File: components/drawer/index.en-US.md [hunk 2]
```diff
@@ -51,10 +51,9 @@ v5 use `rootClassName` & `rootStyle` to config wrapper style instead of `classNa
 | autoFocus | Whether Drawer should get focused after open | boolean | true | 4.17.0 |
 | afterOpenChange | Callback after the animation ends when switching drawers | function(open) | - |  |
 | className | Config Drawer Panel className. Use `rootClassName` if want to config top dom style | string | - |  |
-| classNames | Config Drawer build-in module's className | `header?: string; body?: string; footer?: string; mask?: string; wrapper?: string;` | - |  |
-| styles | Config Drawer build-in module's style | `header?: CSSProperties; body?: CSSProperties; footer?: CSSProperties; mask?: CSSProperties; wrapper?: CSSProperties;` | - | 5.10.0 |
+| classNames | Config Drawer build-in module's className | `header?: string; body?: string; footer?: string; mask?: string; content?: string; wrapper?: string;` | - |  |
+| styles | Config Drawer build-in module's style | `header?: CSSProperties; body?: CSSProperties; footer?: CSSProperties; mask?: CSSProperties; content?: CSSProperties; wrapper?: CSSProperties;` | - | 5.10.0 |
```
Comment (zombieJ) [inline components/drawer/index.en-US.md:55]: @MadCcc 感觉上次说的语义化推导要搞个lint，容器类组件基本上都差不多。怕久了会分叉。

**Review by zombieJ — DISMISSED** [review]

**Comment by afc163** [comment]
现在的 diff 符合预期了。

**Pull request merged**